### PR TITLE
Make GramMatrix losses compatible with unsupervised training

### PR DESCRIPTION
**Pull request by ClementWalter**

### CODE REVIEW

File: keras_fsl/losses/gram_matrix_losses.py [hunk 1]
```diff
@@ -27,38 +36,50 @@ def class_consistency_loss(y_true, y_pred):
         tf.matmul(y_true, tf.matmul(y_pred, y_true), transpose_a=True)[class_mask], class_mask, axis=1
     )
     identity_matrix = tf.eye(tf.shape(confusion_matrix)[0])
-    return tf.reduce_mean(K.binary_crossentropy(identity_matrix, confusion_matrix))
+    return K.binary_crossentropy(identity_matrix, confusion_matrix)
+
 
+class ClassConsistencyLoss(Loss):
+    def call(self, y_true, y_pred):
+        return class_consistency_loss(y_true, y_pred)
```
Comment (Wirg) [inline keras_fsl/losses/gram_matrix_losses.py:44]: pas du tout sur de l'intéret, mais pour la curiosité: 
```python

def print_args(*args):
    print(args)


class Test:
    call = staticmethod(print_args)

if __name__ == '__main__':
    Test().call(5, 7, 7)
```

Comment (Wirg) [reply keras_fsl/losses/gram_matrix_losses.py:44]: du coup. je suis allé voir ce que fait tf, ils ont une sous classe : `LossFunctionWrapper` et derrière :


```python
@keras_export('keras.losses.MeanSquaredError')
class MeanSquaredError(LossFunctionWrapper):
  """Computes the mean of squares of errors between labels and predictions.

  `loss = square(y_true - y_pred)`

  Usage:

  ```python
  mse = tf.keras.losses.MeanSquaredError()
  loss = mse([0., 0., 1., 1.], [1., 1., 1., 0.])
  print('Loss: ', loss.numpy())  # Loss: 0.75
  ```

  Usage with the `compile` API:

  ```python
  model = tf.keras.Model(inputs, outputs)
  model.compile('sgd', loss=tf.keras.losses.MeanSquaredError())
  ```
  """

  def __init__(self,
               reduction=losses_utils.ReductionV2.AUTO,
               name='mean_squared_error'):
    super(MeanSquaredError, self).__init__(
        mean_squared_error, name=name, reduction=reduction)


```

Je pense pas qu'on y gagne grand chose mais à avoir en tête si ça arrive pas mal

**Comment by Wirg** [comment]
Du coup, j'ai fait un test pour les metriques : 
```python

import numpy as np
from tensorflow.keras import Model, layers, losses
from tensorflow.keras.metrics import Accuracy

class AccuracyBis(Accuracy):
    @property
    def name(self):
        return "mycustomname"

x = layers.Input((5,))
model = Model(x, x)

model.compile(loss=losses.binary_crossentropy, metrics=[Accuracy(), AccuracyBis()])

model.fit(np.random.rand(500, 5), np.random.rand(500, 5), epochs=1)


```

Ça marche pas mal.
